app/components/common/eventModal/index.tsx

```diff
@@ -22,26 +22,28 @@ import { patchEvent, postEvent } from '../../../constants/urls.constants';
 import { parseEventToCreateOrUpdateEventPayload, parseEvents } from '../../../utils/event.utils';
 
 interface EventModalProps {
-  events: CalEvent[];
-  currentEvent: CalEvent;
-  createEvent?: (event: CalEvent) => void;
-  setCalendarEvent: React.Dispatch<React.SetStateAction<CalendarEventProps>>;
+  event: CalEvent;
 }
 // This function lets apps optimize which routes data should be
 // reloaded after actions and for client-side navigations.
 // https://remix.run/docs/en/main/route/should-revalidate
 export const unstableShouldReload: ShouldRevalidateFunction = () => false;
 
-export default function EventModal({ events, currentEvent, setCalendarEvent }: EventModalProps) {
-  const { updateEvent, addEvent, view } = useStore((state) => state);
-  const params = useParams();
-  const navigate = useNavigate();
-  const [statuses, setStatuses] = React.useState<{
+export default function EventModal({ event }: EventModalProps) {
+  const { updateEvent, addEvent, view, events } = useStore((state) => state);
+  const [calendarEvent, setCalendarEvent] = useState<CalendarEventProps>({
+    event,
+  });
+  const [statuses, setStatuses] = useState<{
     creatingPost: 'idle' | 'loading';
   }>({
     creatingPost: 'idle',
   });
+  const params = useParams();
+  const navigate = useNavigate();
   const routerLocation = useLocation() as { state: { start: string; end: string } };
+
+  const currentEvent = calendarEvent.event!;
```

Comment:
can this break?

Reply:
> can this break?

@isVivek99 EventModal needs the `event` prop and we set it in calendarEvent initially. So no, unless you manually remove the `event` property from the state, it won't break.

Reply:
can we remove that `!` mark ?

Reply:
@isVivek99 no, its set as an optional in the interface, so its required.

Reply:
on line `25`, i can see it as not being optional